src/wp-includes/class-wp-block-pattterns-registry.php

```diff
@@ -0,0 +1,158 @@
+	 * @param array  $pattern_properties Array containing the properties of the pattern: label, content.
+	 * @return boolean True if the pattern was registered with success and false otherwise.
+	 */
+	public function register( $pattern_name, $pattern_properties ) {
+		if ( ! isset( $pattern_name ) || ! is_string( $pattern_name ) ) {
+			$message = __( 'Pattern name must be a string.' );
+			_doing_it_wrong( __METHOD__, $message, '5.5.0' );
+			return false;
+		}
+
+		$this->registered_patterns[ $pattern_name ] = array_merge(
+			$pattern_properties,
+			array( 'name' => $pattern_name )
+		);
+
+		return true;
+	}
+
+	/**
+	 * Unregisters a pattern.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $pattern_name     Pattern name including namespace.
+	 * @return boolean True if the pattern was unregistered with success and false otherwise.
+	 */
+	public function unregister( $pattern_name ) {
+		if ( ! $this->is_registered( $pattern_name ) ) {
+			/* translators: 1: Pattern name. */
+			$message = sprintf( __( 'Pattern "%1$s" not found.' ), $pattern_name );
+			_doing_it_wrong( __METHOD__, $message, '5.5.0' );
+			return false;
+		}
+
+		unset( $this->registered_patterns[ $pattern_name ] );
+
+		return true;
+	}
+
+	/**
+	 * Retrieves an array containing the properties of a registered pattern.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $pattern_name       Pattern name including namespace.
+	 * @return array Registered pattern properties.
+	 */
+	public function get_registered( $pattern_name ) {
+		if ( ! $this->is_registered( $pattern_name ) ) {
+			return null;
+		}
+
+		return $this->registered_patterns[ $pattern_name ];
+	}
+
+	/**
+	 * Retrieves all registered patterns.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array Array of arrays containing the registered patterns properties,
+	 *               and per style.
+	 */
+	public function get_all_registered() {
+		return array_values( $this->registered_patterns );
+	}
+
+	/**
+	 * Checks if a pattern is registered.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $pattern_name       Pattern name including namespace.
+	 * @return bool True if the pattern is registered, false otherwise.
+	 */
+	public function is_registered( $pattern_name ) {
+		return isset( $this->registered_patterns[ $pattern_name ] );
+	}
+
+	/**
+	 * Utility method to retrieve the main instance of the class.
+	 *
+	 * The instance will be created if it does not exist yet.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return WP_Block_Patterns_Registry The main instance.
+	 */
+	public static function get_instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+}
+
+/**
+ * Registers a new pattern.
+ *
+ * @since 5.5.0
+ *
+ * @param string $pattern_name       Pattern name including namespace.
+ * @param array  $pattern_properties Array containing the properties of the pattern.
+ *
+ * @return boolean True if the pattern was registered with success and false otherwise.
```

Comment:
```suggestion
 * @param array  $pattern_properties Array containing the properties of the pattern.
 * @return bool True if the pattern was registered with success and false otherwise.
```

No empty line between @param and @return tags, see https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#1-functions-class-methods.